platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -314,9 +315,15 @@ function ViewerRetrieveStudyData({
     }
   };
 
+  const prevStudyInstanceUids = usePrevious(studyInstanceUids);
+
   useEffect(() => {
-    studyMetadataManager.purge();
-    purgeCancellablePromises();
+    const hasStudyInstanceUidsChanged = !(prevStudyInstanceUids && prevStudyInstanceUids.every(e => studyInstanceUids.includes(e)));
+
```

Comment:
This useffect is missing prevStudyInstanceUids and purgeCancellablePromises in its deps array.  You can add them or remove the array.

Reply:
The hook should not depend on change of `prevStudyInstanceUids` and 
 `purgeCancellablePromises`, its only watching changes at the variable `studyInstanceUids`.